pkg/operator/operator.go

```diff
@@ -194,13 +195,25 @@ func NewOperator() (context.Context, *Operator) {
 		return []string{o.(*v1beta1.NodeClaim).Status.ProviderID}
 	}), "failed to setup nodeclaim provider id indexer")
 	lo.Must0(mgr.GetFieldIndexer().IndexField(ctx, &v1beta1.NodeClaim{}, "spec.nodeClassRef.apiVersion", func(o client.Object) []string {
-		return []string{o.(*v1beta1.NodeClaim).Spec.NodeClassRef.APIVersion}
+		nc := o.(*v1beta1.NodeClaim)
+		if nc.Spec.NodeClassRef == nil {
```

Comment:
Should we validate that Spec is defined?

Reply:
Should we expanded this check for all the indexers?

Reply:
Spec isn't a pointer, this is really just meant to ensure the error we get when this isn't set isn't just a stack trace. For the required, non pointer types if they're unset because you circumvented validation I think it's fine to just have UB. Really I don't think there's anything strictly wrong with how we crash currently, this is just a free QoL upgrade.